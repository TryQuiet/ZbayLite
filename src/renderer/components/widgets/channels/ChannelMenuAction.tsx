import React from 'react'

import { makeStyles } from '@material-ui/core/styles'
import { Grid } from '@material-ui/core'

import dotsIcon from '../../../static/images/zcash/dots-icon.svg'
import IconButton from '../../ui/IconButton'
import MenuAction from '../../ui/MenuAction'
import MenuActionItem from '../../ui/MenuActionItem'
import ConfirmModal from '../channelSettings/ConfirmModal'

const useStyles = makeStyles((theme) => ({
  menuList: {
    padding: `${theme.spacing(1.5)}px 0`
  },
  icon: {
    width: 30,
    height: 30
  },
  sublabel: {
    color: theme.palette.colors.darkGray,
    letterSpacing: 0.4,
    fontSize: 12,
    lineHeight: '18px'
  }
}))

type ChannelMenuActionProps = {
  onInfo: React.MouseEventHandler<HTMLLIElement>
  onMute: React.MouseEventHandler<HTMLLIElement>
  onUnmute: React.MouseEventHandler<HTMLLIElement>
  onDelete: () => void
  onSettings: () => void
  mutedFlag: boolean
  disableSettings: boolean
  notificationFilter: string
  openNotificationsTab: () => void
}

export const ChannelMenuAction: React.FC<ChannelMenuActionProps> = ({
  onInfo,
  onMute,
  onUnmute,
  onDelete,
  onSettings,
  mutedFlag,
  disableSettings,
  notificationFilter,
  openNotificationsTab
}) => {
  const [openDialog, setOpenDialog] = React.useState(false)
  const classes = useStyles({})
  return (
    <MenuAction
      icon={dotsIcon}
      iconHover={dotsIcon}
      IconButton={IconButton}
      offset='0 8'
    >
      <MenuActionItem onClick={onInfo} title='Info & Invites' />

      <MenuActionItem
        onClick={e => {
          e.preventDefault()
          setOpenDialog(true)
        }}
        closeAfterAction={false}
        title='Remove'
      />

      {!disableSettings ? (
        <MenuActionItem onClick={onSettings} title='Settings' />
      ) : (
        <span />
      )}
      {!disableSettings ? (
        <MenuActionItem
          onClick={() => {
            openNotificationsTab()
            onSettings()
          }}
          title={
            <Grid container direction='column'>
              <Grid item>Notifications</Grid>
              <Grid item className={classes.sublabel}>
                {notificationFilter}
              </Grid>
            </Grid>
          }
        />
      ) : (
        <span />
      )}
      <MenuActionItem
        onClick={mutedFlag ? onUnmute : onMute}
        title={mutedFlag ? 'Unmute' : 'Mute'}
      />
      <ConfirmModal
        open={openDialog}
        title={'Are you sure you want to remove this channel?'}
        actionName='Yes'
        cancelName='No'
        handleClose={() => setOpenDialog(false)}
        handleAction={onDelete}
      />
    </MenuAction>
  )
}

export default ChannelMenuAction
