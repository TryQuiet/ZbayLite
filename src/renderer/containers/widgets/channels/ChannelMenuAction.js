import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { withRouter } from 'react-router-dom'
import ChannelMenuAction from '../../../components/widgets/channels/ChannelMenuAction'
import { actionCreators } from '../../../store/handlers/modals'
import importedChannelHandler from '../../../store/handlers/importedChannel'
import dmChannelSelectors from '../../../store/selectors/directMessageChannel'
import channelSelectors from '../../../store/selectors/channel'
import notificationCenterSelectors from '../../../store/selectors/notificationCenter'
import publicChannelsSelectors from '../../../store/selectors/publicChannels'
import notificationCenterHandlers from '../../../store/handlers/notificationCenter'
import { notificationFilterType } from '../../../../shared/static'
import appHandlers from '../../../store/handlers/app'

const filterToText = {
  [notificationFilterType.ALL_MESSAGES]: 'Every new message',
  [notificationFilterType.MENTIONS]: 'Just @mentions',
  [notificationFilterType.NONE]: 'Nothing',
  [notificationFilterType.MUTE]: 'Muted'
}
export const mapStateToProps = state => {
  return {
    targetAddress: dmChannelSelectors.targetRecipientAddress(state),
    isOwner: channelSelectors.isOwner(state),
    publicChannels: publicChannelsSelectors.publicChannels(state),
    channel: channelSelectors.data(state) || {},
    mutedFlag:
      notificationCenterSelectors.channelFilterById(
        channelSelectors.data(state)
          ? channelSelectors.data(state).address
          : 'none'
      )(state) === notificationFilterType.MUTE,
    notificationFilter:
      // eslint-disable-next-line
      filterToText[
        notificationCenterSelectors.channelFilterById(
          channelSelectors.channel(state)
            ? channelSelectors.channel(state).address
            : 'none'
        )(state)
      ]
  }
}

export const mapDispatchToProps = (dispatch, { history }) => {
  return bindActionCreators(
    {
      onInfo: actionCreators.openModal('channelInfo'),
      onMute: () =>
        notificationCenterHandlers.epics.setChannelsNotification(
          notificationFilterType.MUTE
        ),
      onUnmute: () =>
        notificationCenterHandlers.epics.setChannelsNotification(
          notificationFilterType.ALL_MESSAGES
        ),
      onDelete: () => importedChannelHandler.epics.removeChannel(history),
      publishChannel: actionCreators.openModal('publishChannel'),
      onSettings: actionCreators.openModal('channelSettingsModal'),
      openNotificationsTab: () =>
        appHandlers.actions.setModalTab('notifications')
    },
    dispatch
  )
}

export default R.compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(ChannelMenuAction)
