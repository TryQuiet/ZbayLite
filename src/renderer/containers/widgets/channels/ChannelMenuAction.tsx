import React from 'react'
import { useDispatch } from 'react-redux'
import ChannelMenuAction from '../../../components/widgets/channels/ChannelMenuAction'
import appHandlers from '../../../store/handlers/app'
import { useModal } from '../../hooks'
import { ModalName } from '../../../sagas/modals/modals.types'

const useChannelMenuActionActions = () => {
  const dispatch = useDispatch()
  const onDelete = () => { }
  const openNotificationsTab = () => dispatch(appHandlers.actions.setModalTab('notifications'))

  return { onDelete, openNotificationsTab }
}

const useChannelMenuActionData = () => {
  const data = {
    mutedFlag: false,
    notificationFilter: ''
  }
  return data
}

const ChannelMenuActionContainer = () => {
  const channelInfoModal = useModal(ModalName.channelInfo)
  const channelSettingsModal = useModal(ModalName.channelSettingsModal)

  const { openNotificationsTab, onDelete } = useChannelMenuActionActions()
  const { mutedFlag, notificationFilter } = useChannelMenuActionData()

  return (
    <ChannelMenuAction
      onSettings={channelSettingsModal.handleOpen}
      onInfo={channelInfoModal.handleOpen}
      onDelete={onDelete}
      mutedFlag={mutedFlag}
      notificationFilter={notificationFilter}
      openNotificationsTab={openNotificationsTab}
    />
  )
}

export default ChannelMenuActionContainer
