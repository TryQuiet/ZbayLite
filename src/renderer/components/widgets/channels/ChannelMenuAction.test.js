import React from 'react'
import { shallow } from 'enzyme'

import { mockClasses } from '../../../../shared/testing/mocks'
import { ChannelMenuAction } from './ChannelMenuAction'

describe('ChannelMenuAction', () => {
  it('renders component', () => {
    const result = shallow(
      <ChannelMenuAction
        classes={mockClasses}
        onInfo={jest.fn()}
        onMute={jest.fn()}
        onDelete={jest.fn()}
        onUnmute={jest.fn()}
        onSettings={jest.fn()}
        openNotificationsTab={jest.fn()}
        publicChannels={{}}
        channel={{}}
        publishChannel={() => {}}
        isOwner
        mutedFlag
        disableSettings
        notificationFilter={1}

      />
    )
    expect(result).toMatchSnapshot()
  })
})
