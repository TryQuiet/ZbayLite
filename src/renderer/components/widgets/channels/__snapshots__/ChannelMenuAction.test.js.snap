// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMenuAction renders component 1`] = `
<Memo(WithStyles(MenuAction))
  IconButton={[Function]}
  icon="test-file-stub"
  iconHover="test-file-stub"
  offset="0 8"
>
  <Memo(WithStyles(MenuActionItem))
    onClick={[MockFunction]}
    title="Info & Invites"
  />
  <Memo(WithStyles(MenuActionItem))
    closeAfterAction={false}
    onClick={[Function]}
    title="Remove"
  />
  <span />
  <span />
  <Memo(WithStyles(MenuActionItem))
    onClick={[MockFunction]}
    title="Unmute"
  />
  <Memo(WithStyles(ConfirmModal))
    actionName="Yes"
    cancelName="No"
    handleAction={[MockFunction]}
    handleClose={[Function]}
    open={false}
    title="Are you sure you want to remove this channel?"
  />
</Memo(WithStyles(MenuAction))>
`;
